test/cpp/api/inference_mode.cpp

```diff
@@ -0,0 +1,286 @@
+#include <torch/script.h>
+#include <gtest/gtest.h>
+#include <test/cpp/api/support.h>
+
+using namespace torch::autograd;
+
+torch::Tensor functional_op(torch::Tensor& x) {
+  return x + x;
+}
+
+torch::Tensor& inplace_op(torch::Tensor& x) {
+  return x.add_(1);
+}
+
+torch::Tensor view_op(torch::Tensor& x) {
+  return x.view({2, 3});
+}
+
+/*
+  Only the following combos of Autograd & InplaceOrView keys on tensors are valid:
+    - Autograd=true, InplaceOrView=true
+    - Autograd=false, InplaceOrView=false
+*/
+void assert_tensor_dispatch_keys(torch::Tensor& x, bool Autograd, bool InplaceOrView) {
```

Comment:
If these keys are lockstep, wouldn't it be better to just have `bool InferenceMode` in the argument instead?